src/session/session_api.c

```diff
@@ -1238,6 +1238,256 @@ __session_join(
     API_END_RET_NOTFOUND_MAP(session, ret);
 }
 
+/*
+ * __session_range_uri --
+ *     Session handling of the range stat method with objects.
+ */
+static int
+__session_range_uri(
+  WT_SESSION_IMPL *session, const char *uri, uint64_t *row_countp, uint64_t *byte_countp)
+{
+    WT_CKPT ckpt;
+    WT_DECL_RET;
+    WT_TABLE *table;
+    uint64_t row_count;
+    u_int i;
+
+    table = NULL;
+
+    if (WT_PREFIX_MATCH(uri, "file:")) {
+        /* If we didn't find a metadata entry, map that error to ENOENT. */
+        if ((ret = __wt_meta_checkpoint(session, uri, NULL, &ckpt)) != 0)
+            WT_ERR(ret == WT_NOTFOUND ? ENOENT : ret);
+
+        *row_countp += ckpt.row_count;
+        *byte_countp += ckpt.byte_count;
+        __wt_meta_checkpoint_free(session, &ckpt);
+    } else if (WT_PREFIX_SKIP(uri, "table:")) {
+        /* If we didn't find a metadata entry, map that error to ENOENT. */
+        if ((ret = __wt_schema_get_table(session, uri, strlen(uri), false, 0, &table)) != 0)
+            WT_ERR(ret == WT_NOTFOUND ? ENOENT : ret);
+
+        /* Set the number of rows from one column group, sum the column group byte counts. */
+        row_count = 0; /* [-Wconditional-uninitialized] */
+        for (i = 0; i < WT_COLGROUPS(table); i++) {
+            row_count = 0;
+            WT_ERR(
+              __session_range_uri(session, table->cgroups[i]->source, &row_count, byte_countp));
+        }
+        *row_countp = row_count;
+    } else
+        ret = __wt_bad_object_type(session, uri);
+
+err:
+    if (table != NULL)
+        WT_TRET(__wt_schema_release_table(session, &table));
+
+    return (ret);
+}
+
+/*
+ * __session_range_cursor_table --
+ *     Session handling of the range stat method with table cursors.
+ */
+static int
+__session_range_cursor_table(
+  WT_CURSOR_TABLE *start, WT_CURSOR_TABLE *stop, uint64_t *row_countp, uint64_t *byte_countp)
+{
+    WT_DECL_RET;
+    uint64_t row_count;
+    u_int i;
+
+    /* Set the number of rows from one column group, sum the column group byte counts. */
+    row_count = 0; /* [-Wconditional-uninitialized] */
```

Comment:
It feels a bit odd to me that we have to count it multiple times. Should we return the min or max here?

Reply:
I'm assuming all of the column groups have the same number of rows (because they each have some subset of the columns), so we only count that once, but the total number of bytes in all of the column groups is interesting, and so I'm aggregating that. Does that make sense to you?